controllers/volumereplicationgroup_controller.go

```diff
@@ -773,16 +775,29 @@ func (v *VRGInstance) reconcileVRsAsPrimary() bool {
 			continue
 		}
 
-		if err := v.processVRAsPrimary(pvcNamespacedName, log); err != nil {
-			log.Info("Requeuing due to failure in getting or creating VolumeReplication resource for PersistentVolumeClaim",
-				"errorValue", err)
+		// Protect the PV backend data in the storage volume as primary.
+		if err := v.protectPVBackendDataAsPrimary(pvcNamespacedName, log); err != nil {
+			requeue = true
+
+			continue
+		}
 
+		// Protect the PV cluster data (in etcd)
+		if err := v.protectPVClusterData(pvc, log); err != nil {
 			requeue = true
+			// TODO: use requeueAfter time duration.
 
 			continue
 		}
 
-		log.Info("Successfully processed VolumeReplication for PersistentVolumeClaim")
+		// Annotate that PVC protection is complete
+		if err := v.addProtectedAnnotationToPVC(pvc, log); err != nil {
```

Comment:
Please do! 

I believe you may have originally added a "protecting" annotation and then switched it to "protected" annotation, about which I recall asking for some clarification at that time. Do you recall why you dropped the "protecting" annotation?  Did you drop it because the VRG finalizer on the PVC is doing double duty as the "protecting" annotation?

Reply:
Moving the annotation here does not help us optimize the reconcile (which is what the annotation was meant for). As we would check for the annotation and not process steps in `prepareToProtectEligiblePVC` and that continues with moving the annotation here. As a result, it is possibly better left at the original location in the call stack of `prepareToProtectEligiblePVC`.

Each time we reconcile, we will be backing up the PV to the S3 store post this change for all PVCs that are part of the selector. If we are to use the annotation to skip backing up the PV to the S3 store, then I am not sure if we can detect and handle resize requests, hence optimizing this step may not be needed (and would possibly solve the resize issue as well).